installation_and_upgrade/calibration_files_updater.py

```diff
@@ -57,30 +57,41 @@ def connect_to_drive(self):
             == 0
         )
 
-    def __init__(self, instrument_host, username, password):
+    def __init__(self, instrument_host: str, username: str, password: str) -> None:
         self.username_with_domain = f"{instrument_host}\\{username}"
         self.network_location = r"\\{}\c$\Instrument\Settings\config\common".format(instrument_host)
         self.password = password
 
-    def __enter__(self):
+    def __enter__(self) -> git.Repo | None:
         """
-        Returns: A git repository for the remote calibration folder if connection is successful, else None.
+        Returns: A git repository for the
+        remote calibration folder if connection is successful, else None.
         """
         self.disconnect_from_drive()
         return git.Repo(self.network_location) if self.connect_to_drive() else None
 
-    def __exit__(self, *args):
+    def __exit__(self, *args: Any) -> None:  # noqa: ANN401
         self.disconnect_from_drive()
 
 
-def get_instrument_hosts():
+def get_instrument_hosts() -> Generator[str, None, None]:
     """
     Returns: A collection of instrument host names
```

Comment:
Doesn't return "A collection" if it's a generator (I know you didn't change this, but the type hint has made it obvious...)